sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -1332,3 +1349,432 @@ class MultiFileCloudOrcPartitionReader(
+      // copy stripes to the HostMemoryBuffer
+      withResource(outhmb) { _ =>
+        withResource(new HostMemoryOutputStream(outhmb)) { rawOut =>
+          // All stripes are from the same file, so it's safe to use the first stripe's ctx
+          val ctx = stripes(0).ctx
+          withCodecOutputStream(ctx, rawOut) { (outChannel, protoWriter, codecStream) =>
+            // write the stripes including INDEX+DATA+STRIPE_FOOTER
+            stripes.foreach { stripeWithMeta =>
+              val stripe = stripeWithMeta.stripe
+              stripe.infoBuilder.setOffset(offset + rawOut.getPos)
+              copyStripeData(ctx, outChannel, stripe.inputDataRanges)
+              val stripeFooterStartOffset = rawOut.getPos
+              stripe.footer.writeTo(protoWriter)
+              protoWriter.flush()
+              codecStream.flush()
+              stripe.infoBuilder.setFooterLength(rawOut.getPos - stripeFooterStartOffset)
+            }
+          }
+        }
+      }
+      val bytesRead = fileSystemBytesRead() - startBytesRead
+      // the stripes returned has been updated, eg, stripe offset, stripe footer length
+      (stripes, bytesRead)
+    }
+  }
+
+  /**
+   * To check if the next block will be split into another ColumnarBatch
+   *
+   * @param currentBlockInfo current SingleDataBlockInfo
+   * @param nextBlockInfo    next SingleDataBlockInfo
+   * @return true: split the next block into another ColumnarBatch and vice versa
+   */
+  override def checkIfNeedToSplitDataBlock(
+      currentBlockInfo: SingleDataBlockInfo,
+      nextBlockInfo: SingleDataBlockInfo): Boolean = {
+    val schemaNextFile =
+      nextBlockInfo.schema.getFieldNames.asScala
+    val schemaCurrentfile =
+      currentBlockInfo.schema.getFieldNames.asScala
+
+    if (!schemaNextFile.sameElements(schemaCurrentfile)) {
+      logInfo(s"Orc File schema for the next file ${nextBlockInfo.filePath}" +
+        s" doesn't match current ${currentBlockInfo.filePath}, splitting it into another batch!")
+      return true
+    }
+
+    if (currentBlockInfo.dataBlock.ctx.compressionKind !=
+        nextBlockInfo.dataBlock.ctx.compressionKind) {
+      logInfo(s"Orc File compression for the next file ${nextBlockInfo.filePath}" +
+        s" doesn't match current ${currentBlockInfo.filePath}, splitting it into another batch!")
+      return true
+    }
+
+    false
+  }
+
+  /**
+   * Calculate the output size according to the block chunks and the schema, and the
+   * estimated output size will be used as the initialized size of allocating HostMemoryBuffer
+   *
+   * Please be note, the estimated size should be at least equal to size of HEAD + Blocks + FOOTER
+   *
+   * @param blocks a sequence of data block to be evaluated
+   * @param schema schema info
+   * @return Long, the estimated output size
+   */
+  override def calculateEstimatedBlocksOutputSize(
+      filesAndBlocks: LinkedHashMap[Path, ArrayBuffer[DataBlockBase]],
```

Comment:
maybe I'm missing it did we really need to change this interface to take the LinkedHashMap?  It looks like you ignore the Path part of filesAndBlocks below in the foreach?

Reply:
I don't really understand this response, why is it better to changed to LinkedHashMap and pass parameters you aren't using?   If the answer is because the filesAndBlock is already a LinkedHashMap so we don't have to do any transformations, I'm fine with that but you need to say that and I would like to see it documented, otherwise someone is going to come along later and wonder about it

Reply:
Done, added some comments.